bbl_screen-patch/interpose.cpp

```diff
@@ -726,6 +730,52 @@ SWIZZLE(void, _ZN5BDbus4NodeC2ERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESa
     next(self, s, BusType);
 }
 
+SWIZZLE(int, _Z17get_resource_path19bbl_resource_type_tRNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE, int id, std::string &s)
+    static std::shared_ptr<BDbus::Proxy> proxy;
+    
+    std::string override = "";
+    
+    if (!dbusListener.nobe) {
+        printf("get_resource_path: dbusListener not ready yet?  hope we try again later, I guess");
+    } else {
+        if (!proxy) {
+            proxy = dbusListener.nobe->createProxy("x1plus.x1plusd", "/x1plus/settings");
+        }
+        BDbus::Error e;
+        std::string settings = proxy->callMethod("x1plus.settings", "GetSettings", "{}", e);
+        try {
+            json j = json::parse(settings);
+            auto k = j.at("filament.filename").template get<std::string>();
+            printf("get_resource_path: read filement.filename setting from x1plusd of %s\n", k.c_str());
```

Comment:
typo here in the error message.

`s/filement/filament/`

Reply:
Good catch!  Thanks for the close eyes!